rust/module.rs

```diff
@@ -244,18 +301,84 @@ pub fn module(ts: TokenStream) -> TokenStream {
             &param_name,
             &param_description,
         ));
+        let param_type_internal = match param_type.as_ref() {
+            "str" => "*mut kernel::c_types::c_char",
+            "invbool" => "bool",
+            _ => &param_type,
+        };
+        let param_default = match param_type.as_ref() {
+            "str" => format!("b\"{}\0\" as *const _ as *mut kernel::c_types::c_char", param_default),
+            _ => param_default,
+        };
+        let read_func = match (param_type.as_ref(), permissions_are_readonly(&param_permissions)) {
+            ("str", false) => format!(
+                "
+                    fn read<'par: 'val, 'lck: 'val, 'val>(&'par self, lock: &'lck kernel::KParamGuard) -> &'val [u8] {{
```

Comment:
Shouldn't the return be `&'lck`? The value can be mutated after the lock guard is released, right?

Reply:
The value can be changed at any point time, so we want to give users the ability to read the memory only while the lock guard is alive. So IIUC we need to request the module should outlive the lock which should outlive the returned memory, so `'par: 'lck` and `'lck: 'val`, no? Which can then be simplified to return `'lck` directly, is that right?

Reply:
Yeah `'lck` makes more sense. `self` will be `const`, so this ends up evaluating to `'lck` anyway. Yes, the value might be mutated after the lock is released.